Scripts/Hue Wheel AIDA64.lua

```diff
@@ -0,0 +1,159 @@
+-- purpose with or without fee is hereby granted, provided that the above1
+-- copyright notice and this permission notice appear in all copies.
+--
+-- THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+-- REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+-- AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+-- INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+-- LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE
+-- OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+-- PERFORMANCE OF THIS SOFTWARE.
+
+-- cpu, gpu, fan or hue
+local mode = 'hue'
+
+-- Min and Max speed of the CPU fan
+local fan = {700, 1800}
+-- temperatur minimum and max, applies for gpu and cpu
+local temp = {35, 65}
+
+-- Max CPU temperature and color for alarm
+local cpu = {}
+cpu['max_temp'] = 60
+cpu['r'] = 0
+cpu['g'] = 0
+cpu['b'] = 255
+
+-- Max GPU temperature and color for alarm
+local gpu = {}
+gpu['max_temp'] = 70
+gpu['r'] = 0
+gpu['g'] = 255
+gpu['b'] = 0
+
+-- Max MB temperature and color for alarm
+local mb = {}
+mb['max_temp'] = 50
+mb['r'] = 255
+mb['g'] = 255
+mb['b'] = 255
+
+-- HSV color range
+local color_min = 0
+local color_max = 200
+-- Steps for the "hue" mode
+local color_steps = 0.5
+local delay = 500
+
+Lighting.SetStepDuration(511)
+Lighting.SetFlashingSpeed(0)
+Lighting.SetBreathingModeEnabled(false)
+
+-- https://gist.github.com/GigsD4X/8513963
+function HSVToRGB(hue, saturation, value)
+    -- Returns the RGB equivalent of the given HSV-defined color
+    -- (adapted from some code found around the web)
+
+    -- Get the hue sector
+    local hue_sector = math.floor( hue / 60 );
+    local hue_sector_offset = ( hue / 60 ) - hue_sector;
+
+    local p = value * ( 1 - saturation );
+    local q = value * ( 1 - saturation * hue_sector_offset );
+    local t = value * ( 1 - saturation * ( 1 - hue_sector_offset ) );
+
+    local r
+    local g
+    local b
+    if hue_sector == 0 then
+        r, g, b = value, t, p;
+    elseif hue_sector == 1 then
+        r, g, b = q, value, p;
+    elseif hue_sector == 2 then
+        r, g, b = p, value, t;
+    elseif hue_sector == 3 then
+        r, g, b = p, q, value;
+    elseif hue_sector == 4 then
+        r, g, b = t, p, value;
+    else
+        r, g, b = value, p, q;
+    end;
+    r = tonumber(("%x"):format(r * 15):rep(2), 16)
+    g = tonumber(("%x"):format(g * 15):rep(2), 16)
+    b = tonumber(("%x"):format(b * 15):rep(2), 16)
+
+    return r, g, b
+end;
+
+-- Blinks between red and configured hardware-color - the "alarm"
+function Alarm(r, g, b)
+    for i = 1, 8 do
+        Lighting.SetColour(i, 255, 0, 0)
+    end
+    os.sleep(200)
+    for i = 1, 8 do
+        Lighting.SetColour(i, r, g, b)
+    end
+    os.sleep(200)
+end
+
+function Color(value, temp1, temp2)
+    local percent = (value - temp1) / (temp2 - temp1)
```

Comment:
Note that 'percent' will have a negative value if value < temp1, i.e. temp/fan speed lower than the minimum specified. Maybe you want to handle that specific case to have a different colour? Or simply make that 0?

Reply:
Same in the other scripts.

Reply:
Well, the reason the variables are on top is, that the user needs to set them according to their hardware and known temperature habits. We can't know what their minimum idle and maximum loaded temperatures are.

Reply:
Are you confusing this with some other comment elsewhere? I'm not sure I know what you're talking about.

Reply:
No I'm not. So that you can experience the whole "color range", the user needs to define in which range the temperature can be. Otherwise, for example, only just after the boot you would see the "cold colors" on the LEDs and never again. But if the set minimum is the idle temperature of the hardware, the LED colors can still be all colors in the range.

That's why it is needed, that the user needs to set the min and max RPM and temperature of CPU and GPU.

Reply:
But that's not what I'm talking about. All I'm saying is 'percent' can be negative if 'value' is lower than the minimum value, which means the color_max - percent * ... will actually turn into color_max + percent * ...

Reply:
I think the scripts are pretty good right now. If you could just look into this little thing, I'll merge it.